dash/validator_conn_executor.go

```diff
@@ -0,0 +1,321 @@
+package dash
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/tendermint/tendermint/libs/log"
+	"github.com/tendermint/tendermint/libs/rand"
+	"github.com/tendermint/tendermint/libs/service"
+	"github.com/tendermint/tendermint/libs/sync"
+	"github.com/tendermint/tendermint/p2p"
+	"github.com/tendermint/tendermint/types"
+)
+
+// validatorConnExecutorName contains name that is used to represent the ValidatorConnExecutor in BaseService and logs
+const validatorConnExecutorName = "ValidatorConnExecutor"
+
+// iSwitch defines p2p.Switch methods that are used by this Executor.
+// Useful to create a mock  of the p2p.Switch.
+type iSwitch interface {
+	Peers() p2p.IPeerSet
+
+	AddPersistentPeers(addrs []string) error
+	RemovePersistentPeer(addr string) error
+
+	DialPeersAsync(addrs []string) error
+	IsDialingOrExistingAddress(*p2p.NetAddress) bool
+	StopPeerGracefully(p2p.Peer)
+}
```

Comment:
regarding the interface naming, unfortunately, tendermint has different approaches how to name  the interface, for instance, the package `p2p` has `IPeerSet` and `Peer`. same time, accross the whole project I didn't find another interfaces which begins with `I`. Therefore, I would prefer to use `switch` rather than `iSwitch` or as I suggested before, to use `switcher` that is more widely spread in go code